nvtabular/workflow.py

```diff
@@ -591,43 +591,46 @@ def _aggregated_op(gdf, ops):
             gdf = logic(gdf, columns_ctx, cols_grp, target_cols, stats_context)
         return gdf
 
-    def _aggregated_dask_transform(self, transforms):
+    def _aggregated_dask_transform(self, ddf, transforms):
         # Assuming order of transforms corresponds to dependency ordering
-        ddf = self.get_ddf()
         meta = ddf._meta
         for transform in transforms:
             columns_ctx, cols_grp, target_cols, logic, stats_context = transform
             meta = logic(meta, columns_ctx, cols_grp, target_cols, stats_context)
-        new_ddf = ddf.map_partitions(self.__class__._aggregated_op, transforms, meta=meta)
-        self.set_ddf(new_ddf)
+        return ddf.map_partitions(self.__class__._aggregated_op, transforms, meta=meta)
 
-    def exec_phase(self, phase_index, record_stats=True):
+    def exec_phase(self, phase_index, record_stats=True, update_ddf=True):
         """
         Gather necessary column statistics in single pass.
         Execute one phase only, given by phase index
         """
         transforms = []
-        for task in self.phases[phase_index]:
-            op, cols_grp, target_cols, _ = task
-            if isinstance(op, TransformOperator):
-                stats_context = self.stats if isinstance(op, DFOperator) else None
-                logic = op.apply_op
-                transforms.append((self.columns_ctx, cols_grp, target_cols, logic, stats_context))
-            elif not isinstance(op, StatOperator):
-                raise ValueError("Unknown Operator Type")
-
-        # Preform transforms as single dask task (per ddf partition)
+        phases = range(phase_index + 1)
+        for ind in phases:
```

Comment:
This is sorta confusing to me - maybe we should add some comments on whats happening? Before this function only executed a single phase (which is consistent with the docstring) - but now its executing everything after phase_index,  but not phase_index itself?

Reply:
err my bad - i see that its going from phase 0  to phase_index (instead of from phase_index+1 on) . My bad =). Still - I think we should mention why we are we repeating all the previous phases here.

Reply:
Yeah, good point - I'll add some better comments :)

Reply:
Thanks @benfred - Hopefully the new comments are somewhat helpful?